output/cloud/expv2/metrics_client.go

```diff
@@ -18,47 +18,39 @@ import (
 	"go.k6.io/k6/output/cloud/expv2/pbcloud"
 )
 
-type httpDoer interface {
-	Do(*http.Request) (*http.Response, error)
-}
-
 // metricsClient is a Protobuf over HTTP client for sending
 // the collected metrics from the Cloud output
 // to the remote service.
 type metricsClient struct {
-	httpClient httpDoer
+	httpClient *cloudapi.Client
 	logger     logrus.FieldLogger
 	token      string
 	userAgent  string
-
-	pushBufferPool sync.Pool
-	baseURL        string
+	baseURL    string
 }
 
 // newMetricsClient creates and initializes a new MetricsClient.
-func newMetricsClient(logger logrus.FieldLogger, host string, token string) (*metricsClient, error) {
+func newMetricsClient(
+	logger logrus.FieldLogger, host string, token string, timeout time.Duration,
+) (*metricsClient, error) {
 	if host == "" {
 		return nil, errors.New("host is required")
 	}
 	if token == "" {
 		return nil, errors.New("token is required")
 	}
 	return &metricsClient{
-		httpClient: &http.Client{Timeout: 5 * time.Second},
+		httpClient: cloudapi.NewClient(logger, token, host, "v"+consts.Version, timeout),
```

Comment:
Can we just get this from the `output/cloud/output.go` ? And not need to create a new one.